aspnet/mvc/models/formatting.rst

```diff
@@ -1,8 +1,217 @@
+
+Configuring the JSON Formatter
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A common request when formatting responses using JSON is to use JavaScript-style "camel" casing (the first character of each property is lowercase, then uppercase is used to begin each additional word making up the name). Support for camel casing can be added by using the ``AddJsonOptions`` extension method when configuring MVC:
+
+.. code-block:: c#
+
+  services.AddMvc()
+    .AddJsonOptions(o =>
+      o.SerializerSettings.ContractResolver = 
+        new CamelCasePropertyNamesContractResolver());
+
+Replace the default ``SerializerSettings.ContractResolver`` with a new instance of ``CamelCasePropertyNamesContractResolver``.
+
+Adding XML Format Support
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To add support for XML formatting, add the "Microsoft.AspNetCore.Mvc.Formatters.Xml" package to your *project.json*'s list of dependencies.
+
+.. literalinclude:: formatting/sample/src/ResponseFormattingSample/project.json
+  :language: javascript
+  :lines: 4-26
+  :emphasize-lines: 8
+  :dedent: 2
+
+Next, add the ``XmlSerializerOutputFormatter`` to MVC's list of OutputFormatters in *Startup.cs*:
+
+.. literalinclude:: formatting/sample/src/ResponseFormattingSample/Startup.cs
+  :language: c#
+  :lines: 30-39
+  :emphasize-lines: 6
+  :dedent: 8
+
+Alternately, you can just call the ``AddXmlSerializerFormatters`` extension method:
+
+.. code-block:: c#
+  :emphasize-lines: 2
+
+  services.AddMvc()
+    .AddXmlSerializerFormatters();
+
+Once you've added support for XML formatting, your API methods should return the appropriate format based on the request's ``Accept`` header, as this Fiddler example demonstrates:
+
+.. image:: formatting/_static/xml-response.png
+
+You can see at the top of the image that the GET request was made with an ``Accept: application/xml`` header set. The response at the bottom of the image has a ``Content-Type: application/xml`` header, and you can see at the bottom the ``Author`` object has been serialized to XML.
+
+Making the same request, but with ``application/json`` specified in the ``Accept`` header, results in a JSON formatted response:
+
+.. image:: formatting/_static/json-response-fiddler.png
+
+In this screenshot, you can see the request sets a header of ``Accept: application/json`` and the response specifies the same as its ``Content-Type``. The ``Author`` object is shown in the body of the response, in JSON format.
+
+Forcing a Particular Format
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you would like to require the use of a particular response format, without resorting to using a return type like ``JsonResult``, you can apply the ``[Produces]`` filter. Like most :doc:`/mvc/controllers/filters`, this can be applied at the action, controller, or global scope. When you specify it, you provide the content type that should be produced as a string:
```

Comment:
Delete  When you specify it, you provide the content type that should be produced as a string: